tax_code.go

```diff
@@ -0,0 +1,55 @@
+package xinvoice
+
+import "github.com/invopop/gobl/bill"
+
+// StandardSalesTax is the tax type for sales tax applying at the standard rate
+const StandardSalesTax = "S"
+
+// ZeroRatedGoodsTax is the tax type for goods taxable at the zero rate
+const ZeroRatedGoodsTax = "Z"
+
+// TaxExempt indicates the tax type for exempt transactions
+const TaxExempt = "E"
+
+// ReverseCharge indicates the tax type for reverse charge transactions
+const ReverseCharge = "AE"
+
+// NoSalesTaxIntraCommunity indicates no sales tax is shown for intra-community deliveries
+const NoSalesTaxIntraCommunity = "K"
+
+// TaxNotChargedExportOutsideEU indicates tax is not charged due to export outside the EU
+const TaxNotChargedExportOutsideEU = "G"
+
+// OutOfTaxScope indicates transactions outside the tax scope
+const OutOfTaxScope = "O"
+
+// IGIC represents IGIC (Canary Islands)
+const IGIC = "L"
+
+// IPSI represents IPSI (Ceuta/Melilla)
+const IPSI = "M"
+
+// FindTaxCode finds the tax code for the provided bill line.
+// It returns the found tax code.
+func FindTaxCode(line *bill.Line) string {
+	if len(line.Taxes) == 0 {
+		return StandardSalesTax
+	}
+	tax := line.Taxes[0]
+
+	switch tax.Rate {
+	case "standard":
+		return StandardSalesTax
+	case "zero":
+		return ZeroRatedGoodsTax
+	}
+
+	switch tax.Category {
+	case "IGIC":
```

Comment:
I'm not sure but those codes are currently reserved to the Spanish regime and would require a change to the German regime to work.

Reply:
Could you provide more details?

Reply:
I'm not sure how exactly regimes are used, but the codes you are trying to use are defined in https://github.com/invopop/gobl/blob/main/regimes/es/tax_categories.go and in XInvoice we handle supporting de and fr regimes so the tax category would need to be supported there first.

Reply:
after that you could also change this logic to use cbc.Key types

Reply:
It all makes sense. I think that if it requires changes to GOBL, maybe it shouldn't block this PR.

Reply:
I'm just thinking if we should keep this part until we are ready handle it and then add it with cbc.Key instead of using strings. But if you think that this is ok for the time being then I won't block it